setuptools/command/build_py.py

```diff
@@ -302,6 +302,19 @@ def _get_platform_patterns(spec, package, src_dir):
             for pattern in raw_patterns
         )
 
+    def get_source_files(self):
+        py_files = [module[-1] for module in self.find_all_modules()]
+
+        possible_stub_files = set(os.path.splitext(f)[0] + ".pyi" for f in py_files)
+        stub_files = [f for f in possible_stub_files if os.path.isfile(f)]
```

Comment:
Probably we can use a lazy generator to collapse the 2 loops into a single one, for example:

```suggestion
        possible_stub_files = (os.path.splitext(f)[0] + ".pyi" for f in py_files)
        stub_files = set(f for f in possible_stub_files if os.path.isfile(f))
```

What do you think? This way you avoid running 2 loops (although in the original approach the second loop is potentially shorter, in the average case I don't think `set` will reduce the amount of iterations considerably).

Reply:
This seems good. I only put the `set` in order to make the function output have no repeats rather than for performance reasons (I don't know if avoiding repeats in the output matters but it just seemed a bit wrong to include the same py.typed file loads of times in the case where there are lots of python files in the directory).